src/traffic_server/CoreUtils.cc

```diff
@@ -172,34 +172,29 @@ CoreUtils::insert_table(intptr_t vaddr1, intptr_t offset1, intptr_t fsize1)
   m.fsize = fsize1;
 #endif
 
-  if (arrayMem.length() == 0) {
-    arrayMem(0);
-    arrayMem[(intptr_t)0].vaddr  = vaddr1;
-    arrayMem[(intptr_t)0].offset = offset1;
-    arrayMem[(intptr_t)0].fsize  = fsize1;
+  if (arrayMem.empty()) {
+    arrayMem.push_back({vaddr1, offset1, fsize1});
   } else {
-    intptr_t index = find_vaddr(vaddr1, arrayMem.length(), 0);
-    if (index == arrayMem.length()) {
-      arrayMem(index);
+    unsigned index = find_vaddr(vaddr1, arrayMem.size(), 0);
```

Comment:
This looks very wrong to me. Why doesn't `arrayMem.push_back` or `emplace_back` suffice in all cases? If access to both ends is really needed, use a `std::dequeue`.